modules/common-fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/common/fs2/Environment.scala

```diff
@@ -93,9 +94,11 @@ final case class Environment[F[_], A](
   enrichments: Ref[F, Environment.Enrichments[F]],
   semaphore: Semaphore[F],
   assetsState: Assets.State[F],
-  httpClient: HttpClient[F],
+  httpClient: Http4sClient[F],
+  remoteAdapterHttpClient: Http4sClient[F],
```

Comment:
```suggestion
```
See the other comment, the `Environment` doesn't need to hold this Http4s client, because `HttpClient` could be hold directly inside a `RemoteAdapter`, and also the `Resource` could be created directly inside `Environment.make` so that the HTTP client is automatically released when `Environment` terminates.